js/views/settingsVw.js

```diff
@@ -758,6 +787,7 @@ module.exports = Backbone.View.extend({
     if(newAddress.name || newAddress.street || newAddress.city || newAddress.state || newAddress.postal_code) {
       if(!newAddress.name || !newAddress.street || !newAddress.city || !newAddress.state || !newAddress.postal_code){
         messageModal.show(window.polyglot.t('errorMessages.saveError'), window.polyglot.t('errorMessages.missingError'));
+        setTimeout(() => $(e.target).removeClass('loading'), 0);
```

Comment:
On the other hand, I don't want to over-engineer this either. Perhaps for the time being we just use setTimeout if we want to remove the class from a non-deferred, otherwise directly remove it from the deferred? We could update saveToApi to follow suit. Any areas that don't use saveToApi could just manually make the adjustments (it's not that much work).

Considered its drawbacks, I'm not completely sure that the global click handler is such a big gain over manually having each view add in the 'loading' class as necessary, but we could see how it plays out. It wouldn't be much work to back-track later on, should we decide to.